recipes/yder/all/patches/1.4.18-0001-shared-static-conan.patch

```diff
@@ -0,0 +1,406 @@
+         endif ()
+ 
+@@ -218,16 +260,16 @@ if (BUILD_YDER_TESTING)
+         set(CMAKE_CTEST_COMMAND ctest -V)
+ 
+         set(TST_DIR ${CMAKE_CURRENT_SOURCE_DIR}/test)
+-        set(LIBS yder ${LIBS} ${CHECK_LIBRARIES} ${ORCANIA_LIBRARIES})
++        set(TEST_LIBS ${yder_lib} Check::Check)
+         if (NOT WIN32)
+             find_package(Threads REQUIRED)
+-            set(LIBS ${LIBS} ${CMAKE_THREAD_LIBS_INIT} m)
++            list(APPEND TEST_LIBS ${CMAKE_THREAD_LIBS_INIT} m)
+         endif ()
+         if (NOT APPLE)
+-            set(LIBS ${LIBS} rt)
++            list(APPEND TEST_LIBS rt)
+         endif ()
+         if (NOT WIN32 AND NOT APPLE)
+-            set(LIBS ${LIBS} ${SUBUNIT_LIBRARIES})
++            list(APPEND TEST_LIBS Subunit::Subunit)
+         endif ()
+ 
+         set(TESTS yder_test)
+@@ -239,8 +281,8 @@ if (BUILD_YDER_TESTING)
+ 
+         foreach (t ${TESTS})
+             add_executable(${t} EXCLUDE_FROM_ALL ${TST_DIR}/${t}.c)
+-            target_include_directories(${t} PUBLIC ${TST_DIR})
+-            target_link_libraries(${t} PUBLIC ${LIBS})
++            target_include_directories(${t} PRIVATE ${TST_DIR})
++            target_link_libraries(${t} PUBLIC ${TEST_LIBS})
+             add_test(NAME ${t}
+                     WORKING_DIRECTORY ${TST_DIR}
+                     COMMAND ${t})
+@@ -269,13 +311,16 @@ configure_file(libyder.pc.in libyder.pc @ONLY)
+ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libyder.pc
+         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+ 
+-set(TARGETS yder)
++set(TARGETS )
++if (BUILD_SHARED)
++    list(APPEND TARGETS yder)
++endif ()
+ if (BUILD_STATIC)
+-    set(TARGETS ${TARGETS} yder_static)
++    list(APPEND TARGETS yder_static)
+ endif ()
+ 
+ if (INSTALL_HEADER)
+-    install(TARGETS ${TARGETS}
++    install(TARGETS ${TARGETS} EXPORT YderExports
+             RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+@@ -285,12 +330,33 @@ if (INSTALL_HEADER)
+     install(FILES README.md
+             DESTINATION ${CMAKE_INSTALL_DOCDIR} COMPONENT runtime)
+ else ()
+-    install(TARGETS ${TARGETS}
++    install(TARGETS ${TARGETS} EXPORT YderExports
+             RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+ endif ()
+ 
++set(YDER_INSTALL_CMAKEDIR_DEFAULT "${CMAKE_INSTALL_LIBDIR}/cmake/Yder")
++if (WIN32 AND NOT MINGW)
++    set(YDER_INSTALL_CMAKEDIR_DEFAULT "cmake")
++endif ()
++set(YDER_INSTALL_CMAKEDIR ${YDER_INSTALL_CMAKEDIR_DEFAULT} CACHE STRING "Location where to install the cmake config files")
++
++install(EXPORT YderExports DESTINATION "${YDER_INSTALL_CMAKEDIR}"
++        NAMESPACE "Yder::"
++        FILE "YderTargets.cmake")
++
++configure_package_config_file(cmake-modules/YderConfig.cmake.in YderConfig.cmake
```

Comment:
We should not be adding install or config names 😕 where does this come from?

Reply:
[A pr I created.](https://github.com/babelouest/yder/pull/23)
It's stalling because the maintainer wants to understand the goal of the `xxConfig.cmake`, `xxConfig-targets.cmake` and `xxConfigVersion.cmake` files (see orcania repo).

Reply:
Since this is not accepted upstream, I do think we should be adding them here 🙈 

We should only expose official targets (its fine if they are added in a later release) but we should not be doing it here first

Reply:
Agreed. I'll mark this pr as draft until upstream makes a move.
The guy in charge is welcoming to the idea, so I have good feelings about this.

Reply:
The pr has been accepted. So please give this pr a final review.